Cabal/src/Distribution/Simple/Errors.hs

```diff
@@ -791,3 +793,6 @@ exceptionMessage e = case e of
       ++ " is required but the version of "
       ++ locationPath
       ++ " could not be determined."
+  CombineObjectFilesRelocatableUnsupported ->
+    "combineObjectFiles: Have been asked to combine object "
+      ++ "files but the configured linker does not support relocations."
```

Comment:
The wording of this error seems wrong. The linker definitely does support relocations (e.g. nonrelocatable output still has dynamic relocations). It is rather that it doesn't support producing relocatable object files.